integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/v2/NextMoveServiceRecordProvider.java

```diff
@@ -0,0 +1,37 @@
+package no.difi.meldingsutveksling.nextmove.v2;
+
+import lombok.RequiredArgsConstructor;
+import no.difi.meldingsutveksling.domain.sbdh.StandardBusinessDocument;
+import no.difi.meldingsutveksling.exceptions.ReceiverDoNotAcceptProcessException;
+import no.difi.meldingsutveksling.nextmove.BusinessMessage;
+import no.difi.meldingsutveksling.serviceregistry.SRParameter;
+import no.difi.meldingsutveksling.serviceregistry.ServiceRegistryLookup;
+import no.difi.meldingsutveksling.serviceregistry.ServiceRegistryLookupException;
+import no.difi.meldingsutveksling.serviceregistry.externalmodel.ServiceRecord;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class NextMoveServiceRecordProvider {
+
+    private final ServiceRegistryLookup serviceRegistryLookup;
+
+    ServiceRecord getServiceRecord(StandardBusinessDocument sbd) {
+        BusinessMessage businessMessage = sbd.getBusinessMessage();
+        try {
+            SRParameter.SRParameterBuilder parameterBuilder = SRParameter.builder(sbd.getReceiverIdentifier());
+
+            sbd.getOptionalConversationId().ifPresent(parameterBuilder::conversationId);
+
+            if (businessMessage.getSikkerhetsnivaa() != null) {
+                parameterBuilder.securityLevel(businessMessage.getSikkerhetsnivaa());
+            }
+            return serviceRegistryLookup.getServiceRecord(
+                    parameterBuilder.build(),
+                    sbd.getProcess(),
+                    sbd.getStandard());
+        } catch (ServiceRegistryLookupException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)